__tests__/format-results-test.ts

```diff
@@ -0,0 +1,83 @@
+import { getTodoStorageDirPath } from '@ember-template-lint/todo-utils';
+import { existsSync, readdirSync } from 'fs';
+import { DirResult, dirSync } from 'tmp'; // eslint-disable-line node/no-unpublished-import
+import { formatResultsAsync } from '../src/format-results';
+import * as mutateErrorsToTodos from '../src/mutate-errors-to-todos';
+import { readJson } from './__utils__/read-json';
+
+describe('format-results', () => {
+  const INITIAL_ENV = process.env;
+
+  let tmpDir: DirResult;
+
+  beforeEach(() => {
+    tmpDir = dirSync({ unsafeCleanup: true });
+    process.stdout.write = jest.fn();
+    process.env = { ...INITIAL_ENV, ESLINT_TODO_DIR: tmpDir.name };
+  });
+
+  afterEach(() => {
+    tmpDir.removeCallback();
+  });
+
+  afterAll(() => {
+    process.env = INITIAL_ENV;
+  });
+
+  it('SHOULD NOT generate a TODO dir with todo files when UPDATE_TODO is set to 0', async () => {
+    process.env.UPDATE_TODO = '0';
+
+    const results = await readJson(
+      require.resolve('./__fixtures__/eslint-with-errors.json')
+    );
+
+    await formatResultsAsync(results);
+
+    const todoDir = getTodoStorageDirPath(tmpDir.name);
+    expect(existsSync(todoDir)).toBe(false);
+    expect(process.stdout.write).toHaveBeenCalledTimes(1);
```

Comment:
The expected stdout is already being tested in the `formatter-test` file. Here I am just check if we are in fact printing the errors. I mocked the stdout.write to prevent the eslint errors to be printed with the jest test results.

Reply:
OK. I'm not a fan of this indirection assertion style, meaning you're not testing an explicitly expected condition, but rather a side effect (that the `write` function was called).

Reply:
I understand, I only added this here because I mocked `process.stdout.write` so the lint results are not printed with jest test results and not because I wanted test that stdout was being called. I decided to remove this assertion since it's really not what I'm testing here. I will create acceptance tests to test the output of the CLI based on our previous conversation to test specifically for the output.